src/components/Header.js

```diff
@@ -1,10 +1,39 @@
-
+import { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { toggleMenu } from "../utils/toggleSlice";
+import { YOUTUBE_SEARCH_API } from "../utils/constants";
 
 const Header = () => {
+  const [searchQuery, setSearchQuery] = useState("");
+  const [suggestions, setSuggestions] = useState([]);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const dispatch = useDispatch();
+
+  function handleToggleButton() {
+    dispatch(toggleMenu());
+  }
+
+  async function getSeachSuggestions() {
+    console.log("search", searchQuery);
+    const data = await fetch(YOUTUBE_SEARCH_API + searchQuery);
+    const json = await data.json();
+    setSuggestions(json[1]);
+  }
+
+  useEffect(() => {
```

Comment:
Try debouncing with [this](https://dmitripavlutin.com/react-throttle-debounce/)

Reply:
I have checkedout the article, looks good! However, I would like to continue with the current method for this project as it implements debouncing from scratch.